### PR TITLE
"Fixes #20446 - reset page param after search"

### PR DESCRIPTION
**Pull request by lizagilman**

### CODE REVIEW

File: webpack/assets/javascripts/foreman_tools.js
Comment (ohadlevy) [inline webpack/assets/javascripts/foreman_tools.js]: won't this always set the page to 1 regardless? afaiu this function is called both from the per_page and the search submit?.

can you write a test to demonstrate the various options?

File: webpack/assets/javascripts/foreman_tools.js
Comment (ohadlevy) [inline webpack/assets/javascripts/foreman_tools.js]: I dont think thats correct, if i have a search term and then I'll do pagination, wont this always keep me on the first page?

IMHO only when the search has been submitted (not if it has a value) then you reset the page number to 1. (or remove the page attribute from the url)

**Comment by ohadlevy** [comment]
btw: I think we need to reset the page number when changing the per page value as well.

I just changed from 20 to 5, and then moved 10 pages or so, and then i change it back to 15 where page 10 didn't exists anymore.

**Comment by lizagilman** [comment]
updated

**Comment by lizagilman** [comment]
[test foreman]

**Comment by ohadlevy** [comment]
@lizagilman tests are failing.

**Comment by lizagilman** [comment]
@ohadlevy minor change.. updated

**Comment by lizagilman** [comment]
@ohadlevy ping ?

File: webpack/assets/javascripts/foreman_tools.js
Comment (tbrisker) [inline webpack/assets/javascripts/foreman_tools.js]: `isSearchForm` means that `element` has id of `search-form`, so that the two conditions are actually checking for the same thing.

File: webpack/assets/javascripts/foreman_tools.js
Comment (tbrisker) [inline webpack/assets/javascripts/foreman_tools.js]: If there isn't a value shouldn't we still set the search to an empty value?

File: webpack/assets/javascripts/foreman_tools.js
Comment (tbrisker) [inline webpack/assets/javascripts/foreman_tools.js]: the first clause here is the same as `!isSearchForm`, and the second I think is redundant - if the uri doesn't have search, there is nothing to remove.

File: webpack/assets/javascripts/foreman_tools.js
Comment (tbrisker) [inline webpack/assets/javascripts/foreman_tools.js]: why do we need to check if page > 1? if it is 1 setting it to 1 is a noop, and it makes the `getQueryStringValue` function redundant.

**Review by tbrisker — CHANGES_REQUESTED** [review]
See comments inline, looks like the code can be simplified a bit.

**Comment by lizagilman** [comment]
waiting for review

**Comment by waldenraines** [comment]
@tbrisker @ohadlevy anything else to be done here before merging?

File: webpack/assets/javascripts/foreman_tools.js
Comment (tbrisker) [reply webpack/assets/javascripts/foreman_tools.js]: I think you missed this comment.

**Comment by lizagilman** [comment]
updated

**Review by waldenraines — APPROVED** [review]
Tested this out and it worked for me, thanks @lizagilman!

**Review by tbrisker — APPROVED** [review]
Works well, thanks @lizagilman !

**Pull request merged**
